plugins/jobtype/test/azkaban/jobtype/TestHadoopSecureSparkWrapper.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2012 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.jobtype;
+
+import java.lang.reflect.Field;
+import java.net.URL;
+import java.util.Collections;
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.apache.log4j.Logger;
+import org.apache.hadoop.conf.Configuration;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestHadoopSecureSparkWrapper {
+  private static Logger logger = Logger.getRootLogger();
+
+  @Test
+  public void testAutoLabeling() {
+    Map<String, String> envs = new TreeMap<String, String>(String.CASE_INSENSITIVE_ORDER);
+    envs.put(HadoopSparkJob.SPARK_NODE_LABELING_ENV_VAR, Boolean.TRUE.toString());
+    envs.put(HadoopSparkJob.SPARK_AUTO_NODE_LABELING_ENV_VAR, Boolean.TRUE.toString());
+    envs.put(HadoopSparkJob.SPARK_DESIRED_NODE_LABEL_ENV_VAR, "test2");
+    envs.put(HadoopSparkJob.SPARK_MIN_MEM_VCORE_RATIO_ENV_VAR, "3");
+    URL url = getClass().getClassLoader().getResource("spark-defaults.conf");
+    envs.put(HadoopSparkJob.SPARK_PROPERTY_FILE_PATH_ENV_VAR, url.getPath());
+    setEnv(envs);
+    Configuration.addDefaultResource("yarn-site.xml");
+    String[] argArray = new String[] {
+        "--conf",
+        "spark.yarn.queue=test",
+        "--conf",
+        "spark.yarn.executor.nodeLabelExpression=test",
+        "--executor-cores",
+        "2",
+        "--executor-memory",
+        "7G"
+    };
+    argArray = HadoopSecureSparkWrapper.handleNodeLabeling(argArray);
+    argArray = HadoopSecureSparkWrapper.removeNullsFromArgArray(argArray);
+    Assert.assertTrue(argArray.length == 6);
+    Assert.assertTrue(argArray[1].contains("test2"));
+  }
+
+  @Test
+  public void testDisableAutoLabeling() {
```

Comment:
It's not easy for me to follow the difference between this test and the one before it. Maybe add additional comments or find a way to highlight the difference?